src/ActionsLinux.cpp

```diff
@@ -0,0 +1,210 @@
+#include <fcntl.h>
+
+#include "Actions.hpp"
+#include "MemAllocHeap.hpp"
+#include "StatCache.hpp"
+#include "BinaryData.hpp"
+#include "MemAllocLinear.hpp"
+
+#if defined(TUNDRA_LINUX)
+
+#include <sys/time.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <linux/fs.h>
+#ifdef FICLONE
+#include <sys/ioctl.h>
+#endif
+
+ExecResult CopyFiles(const FrozenFileAndHash* src_files, const FrozenFileAndHash* target_files, size_t files_count, StatCache* stat_cache, MemAllocHeap* heap)
+{
+    ExecResult result;
+    memset(&result, 0, sizeof(result));
+    char tmpBuffer[1024];
+
+    int in_file = -1, out_file = -1;
+    int temporary_pipe[2] = { -1, -1 };
+    pipe(temporary_pipe);
+    const size_t splice_chunk_size = 4096;
+
+    MemAllocLinear scratch;
+    LinearAllocInit(&scratch, heap, 4096, "CopyFiles scratch memory");
+
+    for(size_t i = 0; i < files_count; ++i)
+    {
+        LinearAllocReset(&scratch);
+
+        const char* src_file = src_files[i].m_Filename;
+        const char* target_file = target_files[i].m_Filename;
```

Comment:
I would call this `dest_file`, to avoid confusion around symlinks.

Reply:
Done.